Automattic/newspack-blocks
# Articles Carousel: Improve accessibility

Improves Articles Carousel accessibility through keyboard focus management and screen reader interactions.

### All Submissions:






### Changes proposed in this Pull Request:

* [x] Add the `@wordpress/a11y` package to use `speak()`
* [x] Add the `@wordpress/escape-html` package to escape slide change announcement
* [x] Uses `<button>` elements instead of `<a role="button">` elements
* [x] Removes inactive slide links from the focus order, so you can't tab to a new slide (must use the arrow or pagination bullets)
* [x] Announce new slide content and "slide _ of _" (if not autoplaying) 
* [x] Sets inactive slides to `aria-hidden`
* [x] Announce Paused/Playing state after button press
* [x] Fix focus loss when pressing Pause / Play button
* [x] Moves Pause/Play buttons to _before_ the slider so it can be stopped sooner by assistive technologies like screen readers (AT)
* [x] Remove "This is the last slide" announcement (and all other not-so-helpful Swiper 
 announcements) that prevents the slide content from being announced.
Closes https://github.com/Automattic/newspack-blocks/issues/436
* [x] Unifies `createSwiper` implementation across frontend and editor (Thanks @WunderBart!)

**Editor with VoiceOver on Safari**
![ezgif com-optimize](https://user-images.githubusercontent.com/967608/81612755-f30e3100-93a2-11ea-8cde-6ca78f0ca5f6.gif)

**Frontend with VoiceOver on Safari**
![ezgif com-optimize (1)](https://user-images.githubusercontent.com/967608/81613006-5304d780-93a3-11ea-81c8-21fc591e8339.gif)


### How to test the changes in this Pull Request:

1. Add an Articles Carousel to a post
2. Tab through the slider
3. It should not switch slides when tabbing
4. All Arrows and Navigation dots should be able to be used with the keyboard
5. Check that the current slide's title and author can be tabbed to
6. Pause/Play buttons should work and not lose focus when activating the button with a keyboard
7. Test the above on Editor side as well (you can move focus into the Editor Carousel by clicking initially into the carousel, then tabbing around)

### Other information:
